src/bsp/src/posix/src/BSP.cpp

```diff
@@ -38,13 +34,14 @@ void exampleTopicPublish(void* param) {
         publisher.publish(msg);
         msg.number++;
 
-        vTaskDelay(loopRate);
+        Task::delay(loopRate);
     }
 }
 
 BSP::BSP() :
     m_rosWatchTask("ros_watch", tskIDLE_PRIORITY + 1, rosWatcher, NULL),
-    m_exampleTopicPublishTask("ros_watch", tskIDLE_PRIORITY + 1, rosWatcher, &m_rosNodeHandle) {}
+    m_exampleTopicPublishTask(
+        "ros_watch", tskIDLE_PRIORITY + 1, exampleTopicPublish, &m_rosNodeHandle) {}
```

Comment:
Why did the function pointer changed here?

Reply:
Because it was starting the same function (`rosWatcher`) twice. An error introduced when you refactored to AbstractTasks